runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -11627,6 +11626,60 @@ static bool inlineIsAssignableFrom(TR::Node *node, TR::CodeGenerator *cg)
    return true;
    }
 
+static TR::Register* checkAssignableEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Register *thisClassReg = cg->evaluate(node->getFirstChild());
+   TR::Register *checkClassReg = cg->evaluate(node->getSecondChild());
+
+   TR::Register *resultReg = cg->allocateRegister();
+   TR::LabelSymbol *helperCallLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *doneLabel = generateLabelSymbol(cg);
+
+   int8_t numOfPostDepConditions = (thisClassReg == checkClassReg)? 2 : 3;
+   TR::RegisterDependencyConditions* deps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, numOfPostDepConditions, cg);
+   deps->addPostCondition(thisClassReg, TR::RealRegister::AssignAny);
+   if (thisClassReg != checkClassReg)
+     {
+     deps->addPostCondition(checkClassReg, TR::RealRegister::AssignAny);
+     }
+   deps->addPostCondition(resultReg, TR::RealRegister::AssignAny);
+
+   generateRIInstruction(cg, TR::InstOpCode::LHI, node, resultReg, 1); // load initial value for result
+   generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, helperCallLabel); // branch to OOL section - will make more sense when inlined tests are added
+   TR_S390OutOfLineCodeSection *outlinedSlowPath = new (cg->trHeapMemory()) TR_S390OutOfLineCodeSection(helperCallLabel, doneLabel, cg);
+   cg->getS390OutOfLineCodeSectionList().push_front(outlinedSlowPath);
+   outlinedSlowPath->swapInstructionListsWithCompilation();
+
+   generateS390LabelInstruction(cg, TR::InstOpCode::label, node, helperCallLabel);
+   resultReg = TR::TreeEvaluator::performCall(node, false, cg);
+
+   generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, doneLabel); // skip over fail label
+   outlinedSlowPath->swapInstructionListsWithCompilation();
+   generateS390LabelInstruction(cg, TR::InstOpCode::label, node, doneLabel, deps);
+   node->setRegister(resultReg);
+
+   return resultReg;
+   }
+
+TR::Register *J9::Z::TreeEvaluator::directCallEvaluator(TR::Node *node, TR::CodeGenerator *cg)
```

Comment:
Why do you need this ? `directCallEvaluator` will call the code-generator's inline direct call function which is where you should check and call checkAssignableFromEvaluator. 

[1]. https://github.com/eclipse-omr/omr/blob/3da49aa3aaed9217fb02d5ab3d069aa7142f89ba/compiler/z/codegen/OMRTreeEvaluator.cpp#L11171